Mon Jan 18 19:32:21 CET 2010  Lennart Kolmodin <kolmodin@gentoo.org>
  * Workaround weird wide character error in build warning
  Without this fix, I get the following error when building:
  Criterion/Analysis.hs:131:2:
      Warning: A do-notation statement discarded a result of type ghc-prim:GHC.Prim.Any.
               Suppress this warning by saying "_ <- note
                                                       "mean is %s (%d iterations)\n" (secs <stderr>: hPutChar: invalid argument (Invalid or incomplete multibyte or wide character)
diff -rN -u old-criterion-0.4.1.0/Criterion/Analysis.hs new-criterion-0.4.1.0/Criterion/Analysis.hs
--- old-criterion-0.4.1.0/Criterion/Analysis.hs	2010-01-18 19:33:24.472531953 +0100
+++ new-criterion-0.4.1.0/Criterion/Analysis.hs	2010-01-18 19:33:24.473532150 +0100
@@ -128,7 +128,7 @@
             -> Criterion Double
 analyseMean a iters = do
   let µ = mean a
-  note "mean is %s (%d iterations)\n" (secs µ) iters
+  _ <- note "mean is %s (%d iterations)\n" (secs µ) iters
   noteOutliers . classifyOutliers $ a
   return µ
 
